addons/mail/static/src/discuss/core/common/discuss_channel_model.js

```diff
@@ -7,15 +7,50 @@ export class DiscussChannel extends Record {
     /** @type {number} */
     id;
     get channel_member_ids() {
-        return this.thread.channel_member_ids;
+        return this.thread?.channel_member_ids;
     }
     get channel_type() {
-        return this.thread.channel_type;
+        return this.thread?.channel_type;
     }
     thread = fields.One("Thread", {
         inverse: "channel",
         onDelete: (r) => r.delete(),
     });
+    correspondent = fields.One("discuss.channel.member", {
+        /** @this {import("models").Thread} */
+        compute() {
+            return this.computeCorrespondent();
+        },
+    });
+    /** @returns {import("models").ChannelMember} */
+    computeCorrespondent() {
+        if (this.channel_type === "channel") {
+            return undefined;
+        }
+        const correspondents = this.correspondents;
+        if (correspondents?.length === 1) {
+            // 2 members chat.
+            return correspondents[0];
+        }
+        if (correspondents?.length === 0 && this.channel_member_ids?.length === 1) {
+            // Self-chat.
+            return this.channel_member_ids[0];
+        }
+        return undefined;
+    }
+    correspondentCountry = fields.One("res.country", {
+        /** @this {import("models").Thread} */
+        compute() {
+            return this.correspondent?.persona?.country_id ?? this.thread?.country_id;
+        },
+    });
+    /** @returns {import("models").ChannelMember[]} */
+    get correspondents() {
+        if (!this.channel_member_ids) {
+            return [];
+        }
+        return this.channel_member_ids?.filter(({ persona }) => persona?.notEq(this.store.self));
```

Comment:
Same remark about filter returning an array and useless early return. Also, the optional chaining is not required as you just:
1. just ensured there were some members
2. channel members is a many relation, so it should default to empty array anyway.